syntax = "proto3";

package file_provider;

message Request {
  CommandType command = 1;
  string path = 2;
  int64 chunk_start = 3;
  int64 chunk_size = 4;

  enum CommandType {
    LIST = 0;
    FETCH = 1;
  }
}

message Response {
  ResultType result = 1;
  // LIST
  repeated string filename = 2;
  // FETCH
  int64 chunk_start = 3;
  // chunk_data.size() implies chunk_size
  bytes chunk_data = 4;

  enum ResultType {
    SUCCESS = 0;
    ERROR = 1;
  }
}